### PR TITLE
Added missing aggregations

### PR DESCRIPTION
**Pull request by chyzas**
date range aggregation, missing aggregation

### CODE REVIEW

File: DSL/Aggregation/DateRangeAggregation.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+     * @param string|null $to
+     *
+     * @return RangeAggregation
+     *
+     * @throws \LogicException
+     */
+    public function addRange($from = null, $to = null)
+    {
+        if ($from === null && $to === null) {
+            throw new \LogicException('Missing range');
```
Comment (trylika) [inline DSL/Aggregation/DateRangeAggregation.php]: Exception string could be more informative.

File: DSL/Aggregation/MissingAggregation.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getArray()
+    {
+        if ($this->getField()) {
+            return ['field' => $this->getField()];
+        }
+        throw new LogicException('Missing aggregation must have a field set');
```
Comment (trylika) [inline DSL/Aggregation/MissingAggregation.php]: in other classes you use \LogicException, but here you add LogicException to uses, consistency would be nice :)

File: Tests/Functional/DSL/Aggregation/DateRangeAggregationTest.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+ */
+
+namespace ONGR\ElasticsearchBundle\Tests\Functional\DSL\Aggregation;
+
+use ONGR\ElasticsearchBundle\DSL\Aggregation\DateRangeAggregation;
+use ONGR\ElasticsearchBundle\DSL\BuilderInterface;
+use ONGR\ElasticsearchBundle\ORM\Repository;
+use ONGR\ElasticsearchBundle\Test\AbstractElasticsearchTestCase;
+
+class DateRangeAggregationTest extends AbstractElasticsearchTestCase
```
Comment (trylika) [inline Tests/Functional/DSL/Aggregation/DateRangeAggregationTest.php]: There are no tests that test exception thrown.

File: Tests/Functional/DSL/Aggregation/MissingAggregationTest.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\ElasticsearchBundle\Tests\Functional\DSL\Aggregation;
+
+use ONGR\ElasticsearchBundle\DSL\Aggregation\MissingAggregation;
+use ONGR\ElasticsearchBundle\Test\AbstractElasticsearchTestCase;
+
+class MissingAggregationTest extends AbstractElasticsearchTestCase
```
Comment (trylika) [inline Tests/Functional/DSL/Aggregation/MissingAggregationTest.php]: There are no tests that test exception thrown.

File: Tests/Functional/DSL/Aggregation/DateRangeAggregationTest.php [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        return $out;
+    }
+
+    /**
+     * Test if exception is thrown.
+     *
+     * @expectedException \LogicException
+     * @expectedExceptionMessage Either from or to must be set. Both cannot be null.
+     */
+    public function testIfExceptionIsThrownWhenBothRangesAreNull()
```
Comment (saimaz) [inline Tests/Functional/DSL/Aggregation/DateRangeAggregationTest.php]: This is unit test. It should not exist in functional test.

**Pull request merged**